### PR TITLE
Move validator component to INX plugin

### PR DESCRIPTION
**Pull request by piotrm50**

### CODE REVIEW

File: tools/docker-network/run.sh [hunk 1]
```diff
@@ -39,36 +40,34 @@ fi
 echo $DOCKER_BUILD_CONTEXT $DOCKERFILE_PATH
 docker compose -f $DOCKER_COMPOSE_FILE build --build-arg WITH_GO_WORK=${WITH_GO_WORK:-0} --build-arg DOCKER_BUILD_CONTEXT=${DOCKER_BUILD_CONTEXT} --build-arg DOCKERFILE_PATH=${DOCKERFILE_PATH}
 
-docker compose pull inx-indexer inx-blockissuer inx-faucet
+docker compose pull inx-indexer # inx-blockissuer inx-faucet
```
Comment (muXxer) [inline tools/docker-network/run.sh]: why? :D

File: tools/docker-network/docker-compose.yml [hunk 2]
```diff
@@ -107,64 +97,118 @@ services:
...
-    restart: on-failure
-    environment:
-      - "BLOCKISSUER_PRV_KEY=432c624ca3260f910df35008d5c740593b222f1e196e6cdb8cd1ad080f0d4e33997be92a22b1933f36e26fba5f721756f95811d6b4ae21564197c2bfa4f28270"
-    command:
-      - "--inx.address=validator-1:9029"
-      - "--restAPI.bindAddress=inx-blockissuer:9086"
-      - "--blockIssuer.accountAddress=rms1prkursay9fs2qjmfctamd6yxg9x8r3ry47786x0mvwek4qr9xd9d5c6gkun"
-    networks:
-      - iota-core
+#   inx-blockissuer:
```
Comment (muXxer) [inline tools/docker-network/docker-compose.yml]: Why do you touch the blockissuer and the faucet? Please enable it again.

**Review by muXxer — CHANGES_REQUESTED** [review]

File: tools/docker-network/run.sh (hunk 1 above)
Comment (piotrm50) [reply tools/docker-network/run.sh]: I disabled it temporarily to have cleaner Docker logs and at the time there were some problems with those as they were not compatible with some other changes. It's still a draft so I'll revert it before marking it as ready.

Comment (muXxer) [reply tools/docker-network/run.sh]: Sorry I saw that it's a draft after I finished the review 😅

**Review by muXxer — CHANGES_REQUESTED** [review]
I don't see any changes in the ansible setup.
Can you do these changes there as well?

File: tools/docker-network/docker-compose.yml [hunk 3]
```diff
@@ -245,6 +235,61 @@ services:
...
+      inx-indexer:
+        condition: service_started
+    environment:
+      - "VALIDATOR_PRV_KEY=443a988ea61797651217de1f4662d4d6da11fd78e67f94511453bf6576045a05293dc170d9a59474e6d81cfba7f7d924c09b25d7166bcfba606e53114d0a758b"
+    command:
+      - "--logger.level=debug"
+      - "--inx.address=node-1-validator:9029"
+      - "--validator.ignoreBootstrapped=true"
+      - "--validator.accountAddress=rms1pzg8cqhfxqhq7pt37y8cs4v5u4kcc48lquy2k73ehsdhf5ukhya3y5rx2w6"
+    networks:
```
Comment (muXxer) [inline tools/docker-network/docker-compose.yml:245]: Move the `networks` part before the `environment`.
So we have a default order everywhere.

File: tools/docker-network/docker-compose.yml [hunk 4]
```diff
@@ -245,6 +235,61 @@ services:
...
+        condition: service_started
+      inx-indexer:
+        condition: service_started
+    environment:
+      - "VALIDATOR_PRV_KEY=3a5d39f8b60367a17fd54dac2a32c172c8e1fd6cf74ce65f1e13edba565f281705c1de274451db8de8182d64c6ee0dca3ae0c9077e0b4330c976976171d79064"
+    command:
+      - "--logger.level=debug"
+      - "--inx.address=node-2-validator:9029"
+      - "--validator.accountAddress=rms1pqm4xk8e9ny5w5rxjkvtp249tfhlwvcshyr3pc0665jvp7g3hc875k538hl"
+    networks:
```
Comment (muXxer) [inline tools/docker-network/docker-compose.yml:262]: Move the `networks` part before the `environment`.
So we have a default order everywhere.

File: tools/docker-network/docker-compose.yml [hunk 5]
```diff
@@ -245,6 +235,61 @@ services:
...
+        condition: service_started
+      inx-indexer:
+        condition: service_started
+    environment:
+      - "VALIDATOR_PRV_KEY=db39d2fde6301d313b108dc9db1ee724d0f405f6fde966bd776365bc5f4a5fb31e4b21eb51dcddf65c20db1065e1f1514658b23a3ddbf48d30c0efc926a9a648"
+    command:
+      - "--logger.level=debug"
+      - "--inx.address=node-3-validator:9029"
+      - "--validator.accountAddress=rms1pp4wuuz0y42caz48vv876qfpmffswsvg40zz8v79sy8cp0jfxm4kunflcgt"
+    networks:
```
Comment (muXxer) [inline tools/docker-network/docker-compose.yml:278]: Move the `networks` part before the `environment`.
So we have a default order everywhere.

File: tools/docker-network/docker-compose.yml [hunk 6]
```diff
@@ -245,6 +235,61 @@ services:
...
+    stop_grace_period: 1m
+    restart: unless-stopped
+    depends_on:
+      node-1-validator:
+        condition: service_started
+      inx-indexer:
+        condition: service_started
+    environment:
+      - "VALIDATOR_PRV_KEY=443a988ea61797651217de1f4662d4d6da11fd78e67f94511453bf6576045a05293dc170d9a59474e6d81cfba7f7d924c09b25d7166bcfba606e53114d0a758b"
+    command:
```
Comment (muXxer) [inline tools/docker-network/docker-compose.yml]: Can you use the multiline syntax with `command: >`. Andrea wanted to have it that way for the rest as well.

File: tools/docker-network/docker-compose.yml [hunk 7]
```diff
@@ -245,6 +235,61 @@ services:
...
+    stop_grace_period: 1m
+    restart: unless-stopped
+    depends_on:
+      node-2-validator:
+        condition: service_started
+      inx-indexer:
+        condition: service_started
+    environment:
+      - "VALIDATOR_PRV_KEY=3a5d39f8b60367a17fd54dac2a32c172c8e1fd6cf74ce65f1e13edba565f281705c1de274451db8de8182d64c6ee0dca3ae0c9077e0b4330c976976171d79064"
+    command:
```
Comment (muXxer) [inline tools/docker-network/docker-compose.yml]: Can you use the multiline syntax with command: >. Andrea wanted to have it that way for the rest as well.

File: tools/docker-network/docker-compose.yml [hunk 8]
```diff
@@ -245,6 +235,61 @@ services:
...
+    stop_grace_period: 1m
+    restart: unless-stopped
+    depends_on:
+      node-3-validator:
+        condition: service_started
+      inx-indexer:
+        condition: service_started
+    environment:
+      - "VALIDATOR_PRV_KEY=db39d2fde6301d313b108dc9db1ee724d0f405f6fde966bd776365bc5f4a5fb31e4b21eb51dcddf65c20db1065e1f1514658b23a3ddbf48d30c0efc926a9a648"
+    command:
```
Comment (muXxer) [inline tools/docker-network/docker-compose.yml]: Can you use the multiline syntax with command: >. Andrea wanted to have it that way for the rest as well.

File: deploy/ansible/roles/iota-core-node/templates/docker-compose-iota-core.yml.j2 [hunk 9]
```diff
@@ -125,4 +117,26 @@ services:
...
+
+{% if 'node-01' in inventory_hostname or 'node-02' in inventory_hostname or 'node-03' in inventory_hostname %}
+  inx-validator:
+    container_name: inx-validator
+    image: iotaledger/inx-validator:1.0-alpha
+    stop_grace_period: 1m
+    restart: unless-stopped
+    depends_on:
+      iota-core:
+        condition: service_healthy
```
Comment (muXxer) [inline deploy/ansible/roles/iota-core-node/templates/docker-compose-iota-core.yml.j2]: Isn't that some kind of race condition? Can the nodes even be healthy if there is no validator in the network yet?

**Review by muXxer — APPROVED** [review]

**Review by karimodm — APPROVED** [review]

**Pull request merged**